app/src/main/java/org/mozilla/fenix/components/metrics/MetricsStorage.kt

```diff
@@ -64,27 +77,80 @@ internal class DefaultMetricsStorage(
             Event.GrowthData.FirstUriLoadForDay -> {
                 settings.uriLoadGrowthLastSent = System.currentTimeMillis()
             }
+            Event.GrowthData.FirstWeekSeriesActivity -> {
+                settings.firstWeekSeriesGrowthSent = true
+            }
         }
     }
 
+    private fun updateDaysOfUse() {
+        val daysOfUse = settings.firstWeekDaysOfUseGrowthData
+        val currentDate = Calendar.getInstance()
+        val asString = dateFormatter.format(currentDate.time)
+        if (currentDate.timeInMillis.withinFirstWeek() && daysOfUse.none { it == asString }) {
+            settings.firstWeekDaysOfUseGrowthData = daysOfUse.plus(asString)
+        }
+    }
+
+    private fun shouldTrackFirstWeekActivity(): Boolean = Result.runCatching {
+        if (!System.currentTimeMillis().withinFirstWeek() || settings.firstWeekSeriesGrowthSent) {
+            return false
+        }
+
+        val daysOfUse = settings.firstWeekDaysOfUseGrowthData.map {
+            dateFormatter.parse(it)
+        }.sorted()
+
+        // This loop will check whether the existing list of days of use, combined with the
+        // current date, contains any periods of 3 days of use in a row. It's mostly this ugly
+        // because of the inherent mutability of the Calendar type, but hopefully we can rip
+        // this code out soon™ anyway
+        for (idx in daysOfUse.indices) {
+            if (idx + 1 > daysOfUse.lastIndex || idx + 2 > daysOfUse.lastIndex) {
+                continue
+            }
+            val day1 = daysOfUse[idx]!!.time.toCalendar()
```

Comment:
are the `!!` necessary?

Reply:
According to the compiler 😅 `dateFormatter.parse(it)` returns the platform type `Date!`

Reply:
This is also why this function is wrapped in a `runCatching` block, though

Reply:
I thought about leaving a comment to that effect. Would that be helpful?

Reply:
I admittedly didn't notice the `runCatching` block 😅 

Since the block is wrapped, the `!!` unwrapping is unnecessary, but I could see the argument for keeping them so in the future we know that that operation is volatile. I don't feel super strongly though, especially if this has _actually_ has a short shelf life.